### PR TITLE
Allows using a local image in make create-cluster

### PR DESCRIPTION
**Pull request by sayantani11**
**What type of PR is this?**

**What this PR does / why we need it**:
Allows using a local image in `make create-cluster` to avoid having to push the dev image out to a registry

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Local image usage for `make create-cluster`
```

### CODE REVIEW

**Comment by sayantani11** [comment]
@cpanato I tried using GPG key, but I am kind of unsure about it. If possible can I replace it with SSH Key? Or should we remove this part?

File: Makefile [hunk 1]
```diff
@@ -399,10 +399,10 @@ create-workload-cluster: $(KUSTOMIZE) $(ENVSUBST)
 	$(KUSTOMIZE) build templates | $(ENVSUBST) | kubectl apply -f -
 
 	# Wait for the kubeconfig to become available.
-	${TIMEOUT} 5m bash -c "while ! kubectl get secrets | grep $(CLUSTER_NAME)-kubeconfig; do sleep 1; done"
+	${TIMEOUT} 300m bash -c "while ! kubectl get secrets | grep $(CLUSTER_NAME)-kubeconfig; do sleep 1; done"
 	# Get kubeconfig and store it locally.
 	kubectl get secrets $(CLUSTER_NAME)-kubeconfig -o json | jq -r .data.value | base64 --decode > $(CAPG_WORKER_CLUSTER_KUBECONFIG)
-	${TIMEOUT} 15m bash -c "while ! kubectl --kubeconfig=$(CAPG_WORKER_CLUSTER_KUBECONFIG) get nodes | grep master; do sleep 1; done"
+	${TIMEOUT} 600m bash -c "while ! kubectl --kubeconfig=$(CAPG_WORKER_CLUSTER_KUBECONFIG) get nodes | grep master; do sleep 1; done"
```
Comment (cpanato) [inline Makefile:405]: same here, maybe 20m more

File: hack/create-dev-cluster.sh [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+export GCP_NODE_MACHINE_TYPE="${NODE_MACHINE_TYPE:-n1-standard-2}"
+export WORKER_MACHINE_COUNT=${WORKER_MACHINE_COUNT:-1}
+export KUBERNETES_VERSION="${KUBERNETES_VERSION:-v1.20.9}"
+export CLUSTER_TEMPLATE="${CLUSTER_TEMPLATE:-cluster-template.yaml}"
+
+#identity settings
+export GCP_NETWORK_NAME="default"
+
+# Generate GPG key.
+GPG_KEY_FILE=${GPG_KEY_FILE:-""}
```
Comment (cpanato) [inline hack/create-dev-cluster.sh:50]: I did not get why we need a gpg key?

Comment (sayantani11) [reply hack/create-dev-cluster.sh:50]: I wanted to secure the credentials but capg doesn't need this I guess. I tried ssh key as well but I guess we won't be needing it 🤔

**Comment by sayantani11** [comment]
/retest
We will be needing this for the developer-doc

**Pull request closed without merging**